consent/manager_sql.go

```diff
@@ -287,7 +287,38 @@ func (m *SQLManager) HandleConsentRequest(ctx context.Context, challenge string,
 		strings.Join(sqlParamsConsentRequestHandled, ", "),
 		":"+strings.Join(sqlParamsConsentRequestHandled, ", :"),
 	), d); err != nil {
-		return nil, sqlcon.HandleError(err)
+		err = sqlcon.HandleError(err)
+		if errors.Cause(err) == sqlcon.ErrUniqueViolation {
+			return m.ReplaceUnusedConsentRequest(ctx, challenge, r)
+		}
+		return nil, err
+	}
+
+	return m.GetConsentRequest(ctx, challenge)
+}
+
+func (m *SQLManager) ReplaceUnusedConsentRequest(ctx context.Context, challenge string, r *HandledConsentRequest) (*ConsentRequest, error) {
+	d, err := newSQLHandledConsentRequest(r)
```

Comment:
This method can be shortened, `d` is already defined by it's outer method `HandleConsentRequest` and processing it another time is thus not necessary. `ReplaceUnusedConsentRequest` should also not be exported since it's only being used internally.